sdk/schemaregistry/schema-registry-avro-serializer/src/schemaRegistryAvroSerializer.ts

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { SchemaRegistry } from "@azure/schema-registry";
+import * as avro from "avsc";
+
+// REVIEW: This should go in to a shared doc somewhere that all of the different
+//         language serializer's docs can reference.
+//
+// Wire format
+// -----------
+//
+// This is a standard meant to be reused across schema registry serializers for
+// different format. We only have an avro serializer at this point so picking
+// apart this format is inlined here, but handling could be extracted and shared
+// between serializers of different formats in the future.
```

Comment:
I'm thinking about refactoring the part that handles this envelope from the avro-specific part and making it internal to this lib but ready to be lifted into the main schema-registry package. We decided to keep any abstraction around this out of the first preview as there was a lot of unresolved debate about what such a building block should look like in terms of public surface. I will wait at least until we have a consensus on the public surface of the serializer to do that and maybe until after this PR and first preview, depending on what folks think about it.

Reply:
Tracking this in #10950 for after preview 1.